src/dtos/availability.js

```diff
@@ -0,0 +1,115 @@
+				)
+			)
+		})
+
+		return Object.fromEntries(arrayOfDays)
+	} catch (error) {
+		console.error(error)
+		return false
+	}
+}
+exports.buildUserAvailabilities = ({ startEpoch, endEpoch, userAvailabilities }) => {
+	try {
+		const allEvents = []
+
+		userAvailabilities.forEach((userAvailability) => {
+			if (userAvailability.repeat_unit == null) {
+				// if its a onetime event include it
+				allEvents.push(userAvailability)
+			} else {
+				// if its a recurring event generate events in the date range
+				const start = moment(startEpoch * 1000).startOf('day') // Convert start date to milliseconds and set time to start of day
+				const end = moment(endEpoch * 1000).endOf('day') // Convert end date to milliseconds and set time to end of day
+				let occurrenceDate = moment(userAvailability.start_time * 1000) // Convert event start time to milliseconds and set time to start of day
+				let occurrenceEndDate = moment(userAvailability.end_time * 1000) // Convert event start time to milliseconds and set time to start of day
+
+				while (occurrenceDate.isSameOrBefore(end)) {
+					const occurrenceDateClone = moment(occurrenceDate) // clone of occurrenceDate
+
+					if (
+						occurrenceDate.isSameOrAfter(start) &&
+						(userAvailability.repeat_on == null ||
+							userAvailability.repeat_on.includes(moment.weekdays()[occurrenceDate.day()])) &&
+						(userAvailability.expiration_date == null ||
+							occurrenceDate.isSameOrBefore(moment.unix(userAvailability.expiration_date))) &&
+						(userAvailability.exceptions == null ||
+							userAvailability.exceptions[occurrenceDateClone.startOf('day').unix()] == undefined)
+					) {
+						allEvents.push({
+							...userAvailability,
+							start_time: occurrenceDate.unix(),
+							end_time: occurrenceEndDate.unix(),
+							/* start_time_date: occurrenceDate.utcOffset('+05:30').format('YYYY-MMM-DD hh:mm A'),
+							end_time_date: occurrenceEndDate.utcOffset('+05:30').format('YYYY-MMM-DD hh:mm A'), */
+						})
+					} else if (
+						userAvailability?.exceptions !== null &&
+						userAvailability?.exceptions[occurrenceDateClone.startOf('day').unix()] !== undefined /* &&
+						userAvailability?.exceptions[occurrenceDateClone.startOf('day').unix()]?.start_time !==
+							undefined */
+					) {
+						// if an exception exists for the occurrence date use the updated time from the exception
+						const { start_time, end_time } =
+							userAvailability.exceptions[occurrenceDate.startOf('day').unix()]
+
+						if (start_time !== undefined && end_time !== undefined) {
+							allEvents.push({ ...userAvailability, start_time: start_time, end_time: end_time })
+						}
+					}
+					occurrenceDate.add(
+						userAvailability.repeat_increment || 1,
+						userAvailability.repeat_unit.toLowerCase()
+					)
+					occurrenceEndDate.add(
+						userAvailability.repeat_increment || 1,
+						userAvailability.repeat_unit.toLowerCase()
+					)
+					if (
```

Comment:
@nevil-mathew if repeat_unit is year then?

Reply:
Then we'll add an year to the current date

Reply:
is that logic implemented?